volatility3/framework/plugins/linux/threads.py

```diff
@@ -0,0 +1,67 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+from typing import Callable, Iterable, List, Any
+
+from volatility3.framework import renderers, interfaces
+from volatility3.framework.configuration import requirements
+from volatility3.framework.objects import utility
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.linux import pslist
+
+
+
+
+class Threads(interfaces.plugins.PluginInterface):
+    """Prints threads of a process"""
+
+    _required_framework_version = (2, 0, 0)
```

Comment:
I think `task.get_threads` has been added in `2.0.4`, the minimum required framework version for the plugin is likely to be `2.0.4`.

```suggestion
    _required_framework_version = (2, 0, 4)
```

Reply:
This is important good catch.

Reply:
Thanks, it would've been a great catch if I'd have done the version numbers properly  5:P  Sadly, I only did a patch bump, which doesn't get checked.  I should really have bumped the framework to (2, 1, 0) , and then this should depend on the framework being at least (2, 1, 0).  Sorry I didn't get it quite right, but very good catch @digitalisx, thanks!  5:)

Reply:
I've now bumped the develop branch to `(2, 1, 0)`.

Reply:
Ah, I saw your README change and found out that the version should be `2.0.4`. Thank you again for checking @ikelos, I hope my review is helpful. 🙂

Reply:
> I've now bumped the develop branch to `(2, 1, 0)`.

oops, let me change to 2.1.0
So should `_required_framework_version = (2,1,0)`?

What about `_version` and other parameters like pslist's version? I just want to make sure I understand the versioning methodology correctly.

Reply:
Sure, so roughly it goes as follows:
```Framework <-- _required_framework_version -| plugin <-- _version -| other_plugin```
Your plugin needs the framework to be at least `_required_framework_version` to work, and other plugins can make a `VersionRequirement` on your plugin, saying they need your `_version` to be at least a particular value.  If you haven't changed the way your plugin works, you don't need to do anything.  If you've changed the interface that your plugin exposes, you need to bump MAJOR if it's a breaking change, MINOR if it's an addition and if you just want to tell that a change was made for your records, but it has no external impact, you can bump PATCH.

Hopefully that all makes sense?

Reply:
> Sure, so roughly it goes as follows: `Framework <-- _required_framework_version -| plugin <-- _version -| other_plugin` Your plugin needs the framework to be at least `_required_framework_version` to work, and other plugins can make a `VersionRequirement` on your plugin, saying they need your `_version` to be at least a particular value. If you haven't changed the way your plugin works, you don't need to do anything. If you've changed the interface that your plugin exposes, you need to bump MAJOR if it's a breaking change, MINOR if it's an addition and if you just want to tell that a change was made for your records, but it has no external impact, you can bump PATCH.
> 
> Hopefully that all makes sense?

Okay, I thought so. Just wanted to double-check thanks :)